src/main/scala/com/kinja/amqp/MessageBufferProcessor.scala

```diff
@@ -178,30 +178,31 @@ class MessageBufferProcessor(
 		} yield messages.length
 	}
 
+	private def deleteMessage(msg: MessageLike)(implicit ec: ExecutionContext): Future[Unit] = msg match {
+		case FailedMessage(id, _, _, _, _) =>
```

Comment:
What is your opinion about pattern matching like this:
```
case FailedMesssage(Some(id).....) => messageStore.deleteFailedMessage(....
case FailedMessage(None,......) => Future.failed[Unit](new IllegalStateException("Got a message without an id from database")
```
?

Reply:
I think both are fine. If you'd like it better, I'll change it.

Reply:
I would choose my version because it is not throwing the exception in case of missing id but return with a failed future.

Reply:
Oh, I see. Makes sense.